test/test_jit_fuser_te.py

```diff
@@ -1877,6 +1876,23 @@ def eager(x, y):
                 g = torch.jit.last_executed_optimized_graph()
                 FileCheck().check("TensorExpr").run(g)
 
+    def test_exhaust_specializations(self):
```

Comment:
nit: is there a test where we use multiple strategies? would be great if we could add one.